backend/scripts/import-posts/index.js

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable no-console */
+
+const mongoose = require("mongoose");
+const Airtable = require("airtable");
+require("dotenv").config();
+
+const {
+  config: { mongo, airtable },
+} = require("../../config");
+require("../../lib/models/User");
+const { schema: userSchema } = require("../../lib/models/User");
+const { schema: postSchema } = require("../../lib/models/Post");
+const mapAirtableData = require("./map-airtable-data");
+
+const POSTS_AIRTABLE_NAME = "Posts";
+const SOURCED_BY_FP_USER = {
+  name: "Sourced by FightPandemics",
+};
+const DEFAULT_MAX_RECORDS = 100;
+
+/* TODO: consider library (commander, yargs)
+  For now single positional arg to control records fetched:
+    a) all records: 'npm run import-posts -- -1';
+    b) specify X >= 1: 'npm run import-posts -- X';
+    c) default: 'npm run import-posts' */
+const parseMaxRecordsArg = () => {
+  const maxRecordsArg = process.argv[2];
+  return parseInt(maxRecordsArg, 10) || DEFAULT_MAX_RECORDS;
+};
+
+const importPostsFromAirtable = async (connection, sourcedById) => {
+  const start = Date.now();
+
+  const base = new Airtable({
+    apiKey: airtable.apiKey,
+  }).base(airtable.baseId);
+
+  const Post = connection.model("Post", postSchema);
+  let newPosts = 0;
+  let existingPosts = 0;
+  let failedPosts = 0;
+
+  const selectOptions = { maxRecords: parseMaxRecordsArg() };
+  const records = await base(POSTS_AIRTABLE_NAME).select(selectOptions).all(); // ~12s to get & (locally) import ~2500 posts (~8s to get from Airtable)
+
+  // eslint-disable-next-line no-restricted-syntax
+  for await (const record of records) {
+    try {
+      const res = await Post.findOneAndUpdate(
+        { airtableId: record.id },
+        mapAirtableData(record._rawJson, {
+          id: sourcedById,
+          name: SOURCED_BY_FP_USER.name,
+        }),
+        { new: true, upsert: true, useFindAndModify: false }
+      );
+      res.updatedAt.toString() === res.createdAt.toString() // eslint-disable-line no-unused-expressions
```

Comment:
Maybe we should hide this logic behind a method in the Post schema. Below is an example of that:
```js
// backend/lib/models/Post.js
postSchema.methods.hasBeenUpdated = function hasBeenUpdated() {
  return this.updatedAt.toString() === this.createdAt.toString()
}
```

Reply:
Hi @joaofnds I would agree if we we're using this anywhere else. I think of this script as a rough (hopefully) temporary thing to bootstrap our app so adding functions elsewhere just for it doesn't make sense. If I use bulk write as suggested by Nic I probably will get the update/insert count a different way - http://mongodb.github.io/node-mongodb-native/3.1/api/Collection.html#~BulkWriteOpResult.

I'll try that tomorrow and hopefully I'll have a final PR for approval tomorrow or saturday at the latest.

Reply:
About using bulk, if we are gonna use it I would suggest putting to set the  `updateAt` at the same update doc to keep it consistent.

Something like this
```
update_one: {
  $set: {
    fields...
    updateAt: <<currentTime>>
  }
}
```